python/seldon_core/seldon_methods.py

```diff
@@ -193,19 +197,20 @@ def aggregate(user_model: object, request: prediction_pb2.SeldonMessageList) ->
         request_json = json_format.MessageToJson(request)
         response_json = user_model.aggregate_rest(request_json)
         return json_to_seldon_message(response_json)
-    if hasattr(user_model, "aggregate_grpc"):
+    elif hasattr(user_model, "aggregate_grpc"):
```

Comment:
are you sure this kind of checks will still work if you inherit from SeldonComponent? all attributes will be in the child class

Reply:
Was trying to get round some type check failures. Solved by allowing Any at certain parts  as the user is not required to sub class SeldonComponent at present. Will push an update.

Reply:
I liked the idea of having to instantiate SeldonComponent but then - yes - you can't rely on `hasattr` methods and should use a different mechanism. I reckon you could make it an Interface with no methods, and then branch from it child interfaces with specific set of methods (my guess is that not every method would form a separate interface but you could group some). Users would then inherit several interfaces they need and you would check if their class is an instance of specific interfaces instead of using `hasattr`. This way you could keep strong typing. It sounds more Java, I know, but with type annotations that's where Python seems to be shifting a bit. :-) 

I realise that even if you agree, this a big change, probably requires a bit more thinking, and shouldn't go in this PR. I think using Any is a good solution for now. Not very elegant but works.

Reply:
Yes I think that all makes a lot of sense. We should add an issue when this is merged.